### ISSUE TITLE
Recent commits broke c++ client on Ubuntu

### ISSUE THREAD

**Issue by jedimstr** [TheBlueMatt/RelayNode#24]
That OSX fix pull request broke c++ client on Ubuntu.  I keep getting bitcoind not connected errors.

Reverting my local working copy to commit e4659668f816afa2d3e886bc96aee17b7ad44d35 and make runs fine without bitcoin connect errors.

**Comment by jameshilliard** [comment]
That doesn't make any sense, the only change I made in that commit is wrapped in a #ifdef __APPLE__ I don't see how it could actually change anything related to ubuntu, it seems ok on ubuntu 14.04 x64, got any bitcoind debug logs?

**Comment by jedimstr** [comment]
Nevermind... it was a max connections issue and just "looked" like going back and forth on commits caused bitcoind connection issues.  For multiple iterations of tests between master, your commit, and the old commits I swear it was consistently showing the latest commit failing to connect to bitcoind.  But then I saw same on master, and then after restarting bitcoind, I was able to get your commit to connect.
